platform/android/src/style/sources/custom_geometry_source.cpp

```diff
@@ -90,6 +103,8 @@ namespace android {
 
         // Update the core source
         source.as<mbgl::style::CustomGeometrySource>()->CustomGeometrySource::setTileData(CanonicalTileID(z, x, y), GeoJSON(geometry));
+
+        requestProcessed(z, x, y);
```

Comment:
Why does this method need to be called from the jni bindings. Doesn't it return control to the java code in `GeometryTileRequest.run()` where the request can be marked as processed or completed?

Reply:
Correct, good catch!